app/src/main/java/io/github/sds100/keymapper/system/inputmethod/KeyMapperImeMessenger.kt

```diff
@@ -89,13 +132,57 @@ class KeyMapperImeMessengerImpl(
             return
         }
 
+        // TODO use Android SDK and get version code properly
+        if (Build.VERSION.SDK_INT >= 34) {
+            inputTextRelayService(text, imePackageName)
+        } else {
+            inputTextBroadcast(text, imePackageName)
+        }
+    }
+
+    private fun inputTextBroadcast(text: String, imePackageName: String) {
         Intent(KEY_MAPPER_INPUT_METHOD_ACTION_TEXT).apply {
             setPackage(imePackageName)
 
             putExtra(KEY_MAPPER_INPUT_METHOD_EXTRA_TEXT, text)
             ctx.sendBroadcast(this)
         }
     }
+
+    private fun inputTextRelayService(text: String, imePackageName: String) {
+        // taken from android.view.inputmethod.BaseInputConnection.sendCurrentText()
+
+        if (text.isEmpty()) {
+            return
+        }
+
+        if (text.length == 1) {
+            // If it's 1 character, we have a chance of being
+            // able to generate normal key events...
+            val keyCharacterMap = KeyCharacterMap.load(KeyCharacterMap.VIRTUAL_KEYBOARD)
+
+            val chars = text.toCharArray(startIndex = 0, endIndex = 1)
+
+            val events: Array<KeyEvent> = keyCharacterMap.getEvents(chars)
```

Comment:
Should this one fall back to the other method? Asking because I'm getting nulls here when trying to input an `á`, but, at the same time, I can input `áá`.

Reply:
Can you create an issue so I don't miss this?

Reply:
Yeah, OK: https://github.com/keymapperorg/KeyMapper/issues/1328